x-pack/plugins/security_solution/server/endpoint/services/actions/clients/endpoint_actions_client.ts

```diff
@@ -0,0 +1,215 @@
+      createActionPayload,
+      endpointData,
+    });
+  }
+
+  async isolate(options: IsolationRouteRequestBody): Promise<ActionDetails> {
+    return this.handleResponseAction<IsolationRouteRequestBody, ActionDetails>('isolate', options);
+  }
+
+  async release(options: IsolationRouteRequestBody): Promise<ActionDetails> {
+    return this.handleResponseAction<IsolationRouteRequestBody, ActionDetails>(
+      'unisolate',
+      options
+    );
+  }
+
+  async killProcess(
+    options: KillOrSuspendProcessRequestBody
+  ): Promise<
+    ActionDetails<KillProcessActionOutputContent, ResponseActionParametersWithPidOrEntityId>
+  > {
+    return this.handleResponseAction<
+      KillOrSuspendProcessRequestBody,
+      ActionDetails<KillProcessActionOutputContent, ResponseActionParametersWithPidOrEntityId>
+    >('kill-process', options);
+  }
+
+  async suspendProcess(
+    options: KillOrSuspendProcessRequestBody
+  ): Promise<
+    ActionDetails<SuspendProcessActionOutputContent, ResponseActionParametersWithPidOrEntityId>
+  > {
+    return this.handleResponseAction<
+      KillOrSuspendProcessRequestBody,
+      ActionDetails<SuspendProcessActionOutputContent, ResponseActionParametersWithPidOrEntityId>
+    >('suspend-process', options);
+  }
+
+  async runningProcesses(
+    options: GetProcessesRequestBody
+  ): Promise<ActionDetails<GetProcessesActionOutputContent>> {
+    return this.handleResponseAction<
+      GetProcessesRequestBody,
+      ActionDetails<GetProcessesActionOutputContent>
+    >('running-processes', options);
+  }
+
+  async getFile(
+    options: ResponseActionGetFileRequestBody
+  ): Promise<ActionDetails<ResponseActionGetFileOutputContent, ResponseActionGetFileParameters>> {
+    return this.handleResponseAction<
+      ResponseActionGetFileRequestBody,
+      ActionDetails<ResponseActionGetFileOutputContent, ResponseActionGetFileParameters>
+    >('get-file', options);
+  }
+
+  async execute(
+    options: ExecuteActionRequestBody
+  ): Promise<ActionDetails<ResponseActionExecuteOutputContent, ResponseActionsExecuteParameters>> {
+    return this.handleResponseAction<
+      ExecuteActionRequestBody,
+      ActionDetails<ResponseActionExecuteOutputContent, ResponseActionsExecuteParameters>
+    >('execute', options);
+  }
+
+  async upload(
+    options: UploadActionApiRequestBody
+  ): Promise<ActionDetails<ResponseActionUploadOutputContent, ResponseActionUploadParameters>> {
+    const fleetFiles = await this.options.endpointContext.service.getFleetToHostFilesClient();
+    const fileStream = options.file as HapiReadableStream;
+    const { file: _, parameters: userParams, ...actionPayload } = options;
+    const uploadParameters: ResponseActionUploadParameters = {
+      ...userParams,
+      file_id: '',
+      file_name: '',
+      file_sha256: '',
+      file_size: 0,
+    };
+
+    const createdFile = await fleetFiles.create(fileStream, actionPayload.endpoint_ids);
```

Comment:
We should handle errors for create file failures.

Reply:
There is really nothing we have to do if this fails here, so we just would be re-throwing the error again. The `catch()` block below attempts to delete the file (since they can be large) if there is a failure after it is created/stored (ex. a failure while indexing the documents for the action request).